web3/index.js

```diff
@@ -35,24 +35,87 @@ isIntegrationEnabled()
         loadScripts(config);
       });
 
+      resizeParityBar();
       handleResizeEvents();
     }
   });
 
+function getParityBarElement () {
+  return document.querySelector('#container > div > div');
+}
+
 /**
  *  Propagates opening events to upper frame
  */
 function handleResizeEvents () {
   document.body.addEventListener('parity.bar.visibility', (ev) => {
-    document.querySelector('#container > div > div').style.maxHeight = '100vh';
-    window.parent.postMessage({
+    const { opened } = ev.detail;
+    const parityBarElement = getParityBarElement();
+    const message = {
       type: EV_SIGNER_BAR,
-      opened: ev.detail.opened
-    }, '*');
+      opened
+    };
+
+    parityBarElement.style.maxHeight = '100vh';
+
+    // Resize the iframe if it's closing
+    if (!opened) {
+      return window.setTimeout(() => {
```

Comment:
Would be better to use https://developer.mozilla.org/en-US/docs/Web/Events/transitionend